google-cloud-storage/src/test/java/com/google/cloud/storage/TestUtils.java

```diff
@@ -126,4 +127,24 @@ public ApiClock getClock() {
       }
     };
   }
+
+  /**
+   * Search {@code t} for an instance of {@code T} either directly or via a cause
+   *
+   * @return The found instance of T or null if not found.
+   */
+  @Nullable
+  public static <T extends Throwable> T findThrowable(Class<T> c, Throwable t) {
```

Comment:
Question: wondering why this isn't just a boolean return value if it's being used to check existence of from its usage in this PR?

Reply:
No strong reason. When I wrote the method it felt to make more sense as a things that finds something rather the more "contains"/"caused by" type check.

I can create an overload to return the boolean if you want me to.